File: src/main/java/com/kishlaly/ta/model/AbstractModel.java
package com.kishlaly.ta.model;

import static com.kishlaly.ta.utils.Dates.getBarTimeInMyZone;
import static com.kishlaly.ta.utils.Dates.getTimeInExchangeZone;

public abstract class AbstractModel {

    // in epoch seconds
    protected Long timestamp;

    public static String exchangeTimezome = "US/Eastern";
    protected String nativeDate;
    protected String myDate;

    public AbstractModel(final Long timestamp) {
        this.timestamp = timestamp;
        this.nativeDate = getTimeInExchangeZone(timestamp, exchangeTimezome).toString();
        this.myDate = getBarTimeInMyZone(timestamp, exchangeTimezome).toString();
    }

    public Long getTimestamp() {
        return this.timestamp;
    }
}

File: src/main/kotlin/com/kishlaly/ta/model/AbstractModel.kt
package com.kishlaly.ta.model

import com.kishlaly.ta.utils.Dates

abstract class AbstractModel {

    companion object {
        val exchangeTimezome = "US/Eastern"
    }

    // in epoch seconds
    open public val timestamp: Long
    public val nativeDate: String
    public val myDate: String

    constructor(timestamp: Long) {
        this.timestamp = timestamp
        this.nativeDate = Dates.getTimeInExchangeZone(timestamp, exchangeTimezome).toString()
        this.myDate = Dates.getBarTimeInMyZone(timestamp, exchangeTimezome).toString()
    }

    abstract fun valuesPresent(): Boolean

}